undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -129,12 +129,43 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         if (warMetaData == null) {
             return;
         }
-        String hostName = hostNameOfDeployment(warMetaData, defaultHost);
-        processDeployment(warMetaData, deploymentUnit, phaseContext.getServiceTarget(), hostName);
+        String deploymentName;
+        if (deploymentUnit.getParent() == null) {
+            deploymentName = deploymentUnit.getName();
+        } else {
+            deploymentName = deploymentUnit.getParent().getName() + "." + deploymentUnit.getName();
+        }
+        final UndertowService undertowService = (UndertowService) phaseContext.getServiceRegistry().getService(UndertowService.UNDERTOW).getValue();
+        final Set<Server> servers = undertowService.getServers();
```

Comment:
Does this DeploymentUnitPhaseService depend on all possible Server services? If not there is a race where some Server may not have started and called UndertowService.registerServer.

Same issue applies with respect to all possible Host services, as server.getHosts() at L158 will not be accurate until all Host services have started and called Server.registerHost.